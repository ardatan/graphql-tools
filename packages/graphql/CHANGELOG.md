# @graphql-tools/graphql

## 0.0.1

### Patch Changes

- [#4688](https://github.com/ardatan/graphql-tools/pull/4688) [`72dadfe5`](https://github.com/ardatan/graphql-tools/commit/72dadfe558a18e6ac41da28f5965eb740f4f368e) Thanks [@saihaj](https://github.com/saihaj)! - initial release
