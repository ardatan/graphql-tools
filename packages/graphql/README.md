## `@envelop/graphql`

Fork of GraphQL.js
