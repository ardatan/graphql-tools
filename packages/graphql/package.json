{
  "name": "@graphql-tools/graphql",
  "version": "0.0.1",
  "author": "Saihajpreet Singh <saihajpreet.singh@gmail.com>",
  "license": "MIT",
  "sideEffects": false,
  "repository": {
    "type": "git",
    "url": "https://github.com/ardatan/graphql-tools.git",
    "directory": "packages/graphql"
  },
  "keywords": [
    "gql",
    "graphql",
    "typescript"
  ],
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "exports": {
    ".": {
      "require": {
        "types": "./dist/typings/index.d.cts",
        "default": "./dist/cjs/index.js"
      },
      "import": {
        "types": "./dist/typings/index.d.ts",
        "default": "./dist/esm/index.js"
      },
      "default": {
        "types": "./dist/typings/index.d.ts",
        "default": "./dist/esm/index.js"
      }
    },
    "./*": {
      "require": {
        "types": "./dist/typings/*.d.cts",
        "default": "./dist/cjs/*.js"
      },
      "import": {
        "types": "./dist/typings/*.d.ts",
        "default": "./dist/esm/*.js"
      },
      "default": {
        "types": "./dist/typings/*.d.ts",
        "default": "./dist/esm/*.js"
      }
    },
    "./package.json": "./package.json"
  },
  "typings": "dist/typings/index.d.ts",
  "typescript": {
    "definition": "dist/typings/index.d.ts"
  },
  "devDependencies": {
    "typescript": "4.8.4"
  },
  "buildOptions": {
    "input": "./src/index.ts"
  },
  "publishConfig": {
    "directory": "dist",
    "access": "public"
  },
  "type": "module"
}
