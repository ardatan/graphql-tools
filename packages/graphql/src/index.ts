export * from './error/index.js';
export * from './execution/index.js';
export * from './graphql.js';
export * from './language/index.js';
export * from './type/index.js';
export * from './utilities/index.js';
export * from './validation/index.js';
export * from './version.js';
export * from './subscription/index.js';
