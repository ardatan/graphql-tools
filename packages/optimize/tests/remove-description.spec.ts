import { parse, print } from 'graphql';
import { removeDescriptions } from '../src/index.js';

describe('removeDescription', () => {
  it('should remove descriptions for schemas', () => {
    const doc = parse(/* GraphQL */ `
      """
      test
      """
      type Query {
        """
        something
        """
        f: String
      }

      """
      test
      """
      type Mutation {
        """
        something
        """
        f(
          """
          something
          """
          input: InputType
        ): String
      }

      """
      test
      """
      input InputType {
        """
        something
        """
        f: String
      }

      """
      test
      """
      directive @test on FIELD_DEFINITION

      """
      test
      """
      enum Test {
        """
        something
        """
        A
      }

      """
      test
      """
      interface TestInterface {
        """
        something
        """
        f: String
      }

      """
      test
      """
      union TestUnion = Test | TestInterface

      """
      test
      """
      scalar TestScalar
    `);

    const out = removeDescriptions(doc);
    expect(print(out).trim()).toMatchInlineSnapshot(`
      "type Query {
        f: String
      }

      type Mutation {
        f(input: InputType): String
      }

      input InputType {
        f: String
      }

      directive @test on FIELD_DEFINITION

      enum Test {
        A
      }

      interface TestInterface {
        f: String
      }

      union TestUnion = Test | TestInterface

      scalar TestScalar"
    `);
  });
});
