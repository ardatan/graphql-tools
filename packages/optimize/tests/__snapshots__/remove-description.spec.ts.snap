// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeDescription Should remove description 1`] = `
"type Query {
  f: String
}"
`;
